### ISSUE TITLE
Store Widget ID In State

### ISSUE THREAD

**Issue by enummela** [DataDog/terraform-provider-datadog#301]
Dashboard widget IDs aren't stored in Terraform state, meaning if the widget ID is needed later in the Terraform configuration then there's no straightforward way to obtain it. Would it be possible to add an `id` attribute to the dashboard widget state to assist with retrieving the IDs?

### Terraform Version

```
Terraform v0.12.3
+ provider.datadog v2.2.0
```

### Affected Resource(s)

- datadog_dashboard

### Expected behaviour

The below configuration results in `Success! The configuration is valid.`

```
resource "datadog_dashboard" "ordered_dashboard" {
  title         = "Ordered Layout Dashboard"
  description   = "Created using the Datadog provider in Terraform"
  layout_type   = "ordered"
  is_read_only  = true

  widget {
    alert_graph_definition {
      alert_id = "895605"
      viz_type = "timeseries"
      title = "Widget Title"
      time = {
        live_span = "1h"
      }
    }
  }
}

locals {
  widget_id = datadog_dashboard.ordered_dashboard.widget[0].id
}
```

### Actual Behaviour

```
Error: Unsupported attribute

  on main.tf line 20, in locals:
  20:   widget_id = datadog_dashboard.ordered_dashboard.widget[0].id

This object has no argument, nested block, or exported attribute named "id".
```

**Comment by dabcoder** [comment]
Sorry for the delay in updating this thread. 
Our API allows to get a given widget ID (and the Go client we're using too): https://github.com/zorkian/go-datadog-api/blob/master/board_widgets.go#L46. 

Could you expand on your comment: 

> meaning if the widget ID is needed later in the Terraform configuration

I believe it'd be technically feasible to add it to the TF provider indeed, but it would be good to have a better understanding of your use case first, thanks.

**Comment by enummela** [comment]
@dabcoder thank you for following up.

My use case involves creating a pipeline that will run on a scheduled basis (every 15 minutes or so) that will update the `content` attribute of a note widget with the output of `terraform plan` from a separate Terraform configuration. This would make infrastructure folks aware of possible drift that may occur between their configuration and what is actually applied. If the widget IDs were exposed then I could easily implement using Terraform. Otherwise I need to resort to writing code/scripts.

Something of note I learned while working on this is that the Terraform API does not reliably assign IDs to widgets. I only observed an ID property on a small subset of the widgets returned by the API without any discernible rhyme or reason for these variances. Although maybe this has been fixed in the mean time.
